src/row_objects.ts

```diff
@@ -31,23 +31,23 @@ export class RowObjects<T extends any = Record<string, any>> {
    * protocol. See also:
    * https://exploringjs.com/es6/ch_iteration.html#sec_closing-iterators
    */
-  return(): IteratorResult<T> {
-    return this._rows.return();
+  return(): IteratorResult<Record<string, any>> {
+    let { value, done } = this._rows.return();
+    return { done, value: {} };
   }
 
   /**
    * RowObjects.next
    *
    * Implements the iterator protocol.
    */
-  next(): IteratorResult<T> {
+  next(): IteratorResult<Record<string, any>> {
     const { value, done } = this._rows.next();
     if (done) {
       return { value: value, done: true };
```

Comment:
I think this should not work (`this._rows.next()` should return `any[]`, not `Record<string, any>`, meh TypeScript 😢)